### PR TITLE
fix(HMS-2350): Add a warning if source is unavailable

### PR DESCRIPTION
**Pull request by adiabramovitch**
image

### CODE REVIEW

File: src/Components/SourcesSelect/index.js [hunk 1]
```diff
@@ -65,20 +65,27 @@ const SourcesSelect = ({ setValidation, image }) => {
...
+        // isDisabled={sources?.length === 1}
+        onSelect={onSelect}
+        placeholderText="Select account"
+        aria-label="Select account"
+      >
+        {sources && selectItemsMapper()}
+      </Select>
+      {selected != undefined && sources.find((source) => source.id === selected.id && source.status != 'available') && (
+        <HelperText id="account-error-inline">
+          <HelperTextItem variant="warning">source is unavailable</HelperTextItem>
```
Comment (ezr-ondrej) [inline src/Components/SourcesSelect/index.js:85]: Would we want to include a link to sources, or any other resource to how to fix that?

File: src/Components/SourcesSelect/index.js [hunk 2]
```diff
@@ -65,20 +65,27 @@ const SourcesSelect = ({ setValidation, image }) => {
...
+        maxHeight="180px"
+        // TODO decide if to disable the select
+        // isDisabled={sources?.length === 1}
+        onSelect={onSelect}
+        placeholderText="Select account"
+        aria-label="Select account"
+      >
+        {sources && selectItemsMapper()}
+      </Select>
+      {selected != undefined && sources.find((source) => source.id === selected.id && source.status != 'available') && (
```
Comment (amirfefer) [inline src/Components/SourcesSelect/index.js:83]: nit: For improving readability, could you please extract it to a variable?
```js
const isSelectedSourceInvalid = sources.find((source) => source.id === selected.id && source.status != 'available')
// {...}

{isSelectedSourceInvalid && (
  <HelperText>...</HelperText>
)}

File: src/Components/SourcesSelect/index.js [hunk 3]
```diff
@@ -65,20 +65,27 @@ const SourcesSelect = ({ setValidation, image }) => {
...
-      // isDisabled={sources?.length === 1}
-      onSelect={onSelect}
-      placeholderText="Select account"
-      aria-label="Select account"
-    >
-      {sources && selectItemsMapper()}
-    </Select>
+    <>
+      <Select
+        ouiaId="select_account"
```
Comment (amirfefer) [inline src/Components/SourcesSelect/index.js:70]: can you add `validated=warning` prop if the selected source is invalid?

**Review by amirfefer — DISMISSED** [review]
Thanks @adiabramovitch 👍

File: src/Components/SourcesSelect/index.js (hunk 3 above)
Comment (adiabramovitch) [reply src/Components/SourcesSelect/index.js:70]: I don't see any `validate` prop here: https://www.patternfly.org/components/menus/select/#props

File: src/Components/SourcesSelect/index.js (hunk 1 above)
Comment (adiabramovitch) [reply src/Components/SourcesSelect/index.js:85]: Sure, what text would we want to write here? 
source is unavailable, for more information click [here](https://console.redhat.com/settings/sources)?

Comment (ezr-ondrej) [reply src/Components/SourcesSelect/index.js:87]: yeah, exactly! I'd maybe link to the source details (specific source) directly, but apart from that :+1:

File: src/Components/SourcesSelect/index.js [hunk 4]
```diff
@@ -63,22 +65,38 @@ const SourcesSelect = ({ setValidation, image }) => {
...
+        onSelect={onSelect}
+        placeholderText="Select account"
+        aria-label="Select account"
+      >
+        {sources && selectItemsMapper()}
+      </Select>
+      {isSelectedSourceUnavailable && (
+        <HelperText id="account-error-inline">
+          <HelperTextItem variant="warning">
+            source is unavailable, for more information click <a href={link}> here </a>
```
Comment (amirfefer) [inline src/Components/SourcesSelect/index.js:95]: do we want to open the link in a new window?

Comment (adiabramovitch) [reply src/Components/SourcesSelect/index.js]: Opened it in a new tab , the linter added the spaces like that.

**Review by ezr-ondrej — APPROVED** [review]
This is ready to be merged, but I'd like to wait for the pipeline approval, let see if we can get it green :)

**Comment by ezr-ondrej** [comment]
Thanks @adiabramovitch ! :+1:

**Pull request merged**